src/frame/timestamp.rs

```diff
@@ -59,48 +58,91 @@ impl fmt::Display for Timestamp {
     }
 }
 
-impl str::FromStr for Timestamp {
+struct Parser<'a>(&'a str);
+
+impl Parser<'_> {
+    pub fn parse_timestamp(&mut self, source: &str) -> Result<Timestamp, ()> {
```

Comment:
Could you please remove this `pup`?

Reply:
I was wondering why you wanted me to remove the `Parser` struct, that method or whatever "pup" could have meant :-).

Done.